### ISSUE TITLE
`std_instead_of_core`/etc. miss macros

### ISSUE THREAD

**Issue by bushrat011899** [rust-lang/rust-clippy#17260]
### Summary

`std_instead_of_core` and friends explicitly do not lint for macros. This was done to avoid linting `std::env`, which refers to the module `std::env` *and* the macro `core::env`. However, this also causes other macros, such as `writeln`, `stringify`, etc. to be missed.

#17252 changes how the lints generate suggestions in a way that resolves this issue. All that's missing is the removal of the explicit bail-out for macros:

https://github.com/rust-lang/rust-clippy/blob/d7246b03ea4761416cfba19d34708e4200fce604/clippy_lints/src/std_instead_of_core.rs#L131

Which should be done in a follow-up PR.

### Lint Name

std_instead_of_core

### Reproducer

I tried this code:

```rust
use std::writeln;
```

I expected to see this happen:

`std_instead_of_core` should suggest replacing the above with `use core::writeln;`

Instead, this happened:

The lint is not triggered.

### Version

```text
rustc 1.98.0-nightly (485ec3fbc 2026-06-10)
binary: rustc
commit-hash: 485ec3fbcc12fa14ef6596dabb125ad710499c9e
commit-date: 2026-06-10
host: x86_64-pc-windows-msvc
release: 1.98.0-nightly
LLVM version: 22.1.6
```

**Comment by profetia** [comment]
You can claim this issue if you intend to do the follow up.

**Comment by bushrat011899** [comment]
@rustbot claim

Thanks for the reminder! I should've done that from the start.
